### PR TITLE
First version of fix save and restore

### PR DESCRIPTION
**Pull request by mark-todd**

### CODE REVIEW

File: endgame_simulations/endgame_simulation.py [hunk 1]
```diff
@@ -64,12 +69,34 @@ def __init__(
 
         else:
             assert input
-            # input
-            simulation = type(self).simulation_class.restore(input=input)
+            h5 = (
+                input
+                if isinstance(input, (h5py.File, h5py.Group))
+                else h5py.File(input, "w")
```
Comment (mkcinu) [inline endgame_simulations/endgame_simulation.py]: "r"?

Comment (mark-todd) [reply endgame_simulations/endgame_simulation.py]: Good spot!

File: endgame_simulations/endgame_simulation.py [hunk 2]
```diff
@@ -78,10 +105,20 @@ def save(self, output: FileType) -> None:
...
-        self.simulation.save(output)
+        h5 = (
+            output
+            if isinstance(output, (h5py.File, h5py.Group))
+            else h5py.File(output, "w")
+        )
+        grp = h5.create_group("simulation")
+        self.simulation.save(grp)
+        h5.attrs["param_set"] = json.dumps(
+            [(i[0], i[1].dict()) for i in self._param_set]
```
Comment (mkcinu) [inline endgame_simulations/endgame_simulation.py:116]: Why is this needed? Can't you just run self._param_set.json() ?

Comment (mark-todd) [reply endgame_simulations/endgame_simulation.py:116]: self._param_set is a list - list doesn't have that method right?
